packages/app/src/specs/RequestAccessButton.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <Button
+    v-if="projectConnectionStatus === 'UNAUTHORIZED'"
+    :prefix-icon="SendIcon"
+    prefix-icon-class="icon-dark-white icon-light-transparent"
+    data-cy="request-access-button"
+    @click="requestAccess"
+  >
+    {{ t("specPage.requestAccessButton") }}
+  </Button>
+  <Button
+    v-else-if="projectConnectionStatus === 'ACCESS_REQUESTED'"
+    :prefix-icon="SendIcon"
+    prefix-icon-class="icon-dark-white icon-light-transparent"
+    data-cy="access-requested-button"
+    class="bg-gray-800 border-gray-800"
+    disabled
+  >
+    {{ t("specPage.requestSentButton") }}
+  </Button>
+</template>
+
+<script setup lang="ts">
+import Button from '@cy/components/Button.vue'
+import SendIcon from '~icons/cy/paper-airplane_x16.svg'
+import { RequestAccessButtonFragment, RequestAccessButton_RequestAccessDocument } from '../generated/graphql'
+import { useI18n } from '@cy/i18n'
+import { computed } from 'vue'
+import { gql, useMutation } from '@urql/vue'
+
+gql`
+fragment RequestAccessButton on Query {
+  currentProject {
+    id
+    projectId
+    cloudProject {
+      __typename
+      ... on CloudProjectUnauthorized {
+        message
+        hasRequestedAccess
+      }
+    }
+  }
+}
+`
+
+gql`
+mutation RequestAccessButton_RequestAccess( $projectId: String! ) {
+  cloudProjectRequestAccess(projectSlug: $projectId) {
+    __typename
+    ... on CloudProjectUnauthorized {
+      message
+      hasRequestedAccess
+    }
+  }
+}
+`
+
+const { t } = useI18n()
+
+const props = defineProps<{
+  gql: RequestAccessButtonFragment
+}>()
+
+const projectId = computed(() => props.gql.currentProject?.projectId)
```

Comment:
Do we need this computed value? It's not actually deriving anything, right?

Reply:
This is a bad habit on my part - I tend to use `computed` to clean up some of the TS overhead, especially when a deeply-nested conditional field is used in the template. In this case I think the component was refactored so this value can be easily extracted as an inline inside the `requestAccess` function